steel/src/lib.rs

```diff
@@ -32,79 +36,24 @@ mod state;
 pub use contract::{CallBuilder, Contract};
 pub use mpt::MerkleTrie;
 
-/// The serializable input to derive and validate a [EvmEnv].
+/// The serializable input to derive and validate an [EvmEnv] from.
 #[derive(Clone, Serialize, Deserialize)]
-pub struct EvmInput<H> {
-    header: H,
-    state_trie: MerkleTrie,
-    storage_tries: Vec<MerkleTrie>,
-    contracts: Vec<Bytes>,
-    ancestors: Vec<H>,
+pub enum EvmInput<H> {
```

Comment:
Should we mark this as `#[non_exhaustive]`, which should theoretically allow us to add more input types in the future as a non-breaking change.

Reply:
Good idea. I went ahead and did that in https://github.com/risc0/risc0-ethereum/pull/180/commits/9fe9316085da9004ad5e11914417a22c6e4961f4